src/atsu/src/com/webinterfaces/NXApiConnector.ts

```diff
@@ -201,7 +201,7 @@ export class NXApiConnector {
             }
 
             // Fetch new messages
-            Telex.fetchMessages()
+            await Telex.fetchMessages()
                 .then((data) => {
```

Comment:
Something like:

```
// Fetch new messages
            try {
                const data = await Telex.fetchMessages();
                for (const msg of data) {
                    const message = new FreetextMessage();
                    message.Network = AtsuMessageNetwork.FBW;
                    message.Direction = AtsuMessageDirection.Uplink;
                    message.Station = msg.from.flight;
                    message.Message = msg.message.replace(/;/i, ' ');

                    retval.push(message);
                }
            } catch (e) {
                return [AtsuStatusCodes.ComFailed, retval];
            }
        }
```

Reply:
I don't see a real difference. Your version waits for the result of fetchMessages() and my version is waiting for the resulting array. Isn't it the same?

Reply:
I was just under the impression that either "await" or ".then" is being used. But I guess both works.

There could be an "issue" because as a result of "await" I want the result of the async operation. If I use ".then" in the async operation, the return value will be  ``undefined``.
 
It's not an issue now but just general practice I think. I won't block the PR due to this 👍